ui/src/app/base/components/ScriptStatus/ScriptStatus.js

```diff
@@ -0,0 +1,108 @@
+import PropTypes from "prop-types";
+import React from "react";
+
+import { scriptStatus } from "app/base/enum";
+import Tooltip from "app/base/components/Tooltip";
+
+const ScriptStatus = ({
+  children,
+  hidePassedIcon = false,
+  scriptType,
+  tooltipPosition
+}) => {
+  switch (scriptType.status) {
+    case scriptStatus.PASSED: {
+      return hidePassedIcon ? (
+        children
+      ) : (
+        <>
+          <i className="p-icon--success is-inline" />
+          {children}
+        </>
+      );
+    }
+
+    case scriptStatus.DEGRADED:
+    case scriptStatus.FAILED:
+    case scriptStatus.FAILED_APPLYING_NETCONF:
+    case scriptStatus.FAILED_INSTALLING: {
+      return (
+        <Tooltip message="Machine has failed tests." position={tooltipPosition}>
+          <i className="p-icon--error is-inline" />
+          {children}
+        </Tooltip>
+      );
+    }
+
+    case scriptStatus.TIMEDOUT: {
+      return (
+        <Tooltip
+          message="Machine has tests that have timed out."
+          position={tooltipPosition}
+        >
+          <i className="p-icon--timed-out is-inline" />
+          {children}
+        </Tooltip>
+      );
+    }
+
+    case scriptStatus.APPLYING_NETCONF:
+    case scriptStatus.INSTALLING:
+    case scriptStatus.RUNNING: {
+      return (
+        <>
+          <i className="p-icon--running is-inline" />
+          {children}
+        </>
+      );
+    }
+
+    case scriptStatus.PENDING: {
```

Comment:
As mentioned, this gets a tooltip currently, although I think the tooltip is possibly wrong in the angular app, and should indicate that this is pending?

Reply:
@squidsoup Yeah I'm pretty sure the angular tooltip is wrong because it assumes any non-passing status == failed. I opted to just remove the tooltip entirely if the status is pending or running because I figured the icon was enough, but maybe my UX-fu is off.

@lilyvidenova any thoughts on if tooltips should show for machines with pending/running tests, and if so what should they say?

Reply:
@lilyvidenova oh yeah also I added a warning icon and tooltip if a machine hasn't run any tests of that type yet ("Machine has not run any tests of this type.")